var/spack/repos/builtin/packages/gpi-2/package.py

```diff
@@ -3,77 +3,176 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from os import environ
 
 from spack import *
 
 
 class Gpi2(AutotoolsPackage):
-    """GPI-2 is an API for the development of scalable, asynchronous and fault
-    tolerant parallel applications. It implements the GASPI specification
-    (www.gaspi.de)"""
+    """GPI-2 implements GASPI specification, an API specification for
+    asynchronous communication. It provides a flexible, scalable and
+    fault tolerant interface for parallel applications.
+    """
 
-    homepage = "http://www.gpi-site.com"
-    url      = "https://github.com/cc-hpc-itwm/GPI-2/archive/v1.4.0.tar.gz"
-    git      = "https://github.com/cc-hpc-itwm/GPI-2.git"
-    maintainers = ['robert-mijakovic']
+    homepage = 'http://www.gpi-site.com'
+    url      = 'https://github.com/cc-hpc-itwm/GPI-2/archive/refs/tags/v1.5.1.tar.gz'
+    git      = 'https://github.com/cc-hpc-itwm/GPI-2.git'
+
+    maintainers = ['robert-mijakovic', 'acastanedam']
 
     version('develop', branch='next')
     version('master', branch='master')
+
+    version('1.5.1', sha256='4dac7e9152694d2ec4aefd982a52ecc064a8cb8f2c9eab0425428127c3719e2e')
+    version('1.5.0', sha256='ee299ac1c08c38c9e7871d4af745f1075570ddbb708bb62d82257244585e5183')
     version('1.4.0', sha256='3b8ffb45346b2fe56aaa7ba15a515e62f9dff45a28e6a014248e20094bbe50a1')
+    version('1.3.2', sha256='83dbfb2e4bed28ef4e2ae430d30505874b4b50252e2f31dc422b3bc191a87ab0')
+    version('1.3.1', sha256='414fa352e7b478442e6f5d0b51ff00deeb4fc705de805676c0e68829f3f30967')
+    version('1.3.0', sha256='ffaa5c6abfbf79aec6389ab7caaa2c8a91bce24fd046d9741418ff815cd445d2')
+    version('1.2.0', sha256='0a1411276aa0787382573df5e0f60cc38ca8079f2353fb7a7e8dc57050a7d2cb')
+    version('1.1.0', sha256='626727565a8b78be0dc8883539b01aaff2bb3bd42395899643bc4d6cc2313773')
+    version('1.0.2', sha256='b03b4ac9f0715279b2a5e064fd85047cb640a85c2361d732930307f8bbf2aeb8')
+    version('1.0.1', sha256='b1341bb39e7e70334d7acf831fe7f2061376e7516b44d18b31797748c2a169a3')
+
+    depends_on('gawk')
+
+    variant('fortran', default=False, description='Enable Fortran modules')
+
+    variant(
+        'mpi',
+        values=disjoint_sets(
+            ('openmpi',), ('mpich',), ('mvapich',),
+        ).with_non_feature_values('none'),
+        description='List of MPIs',
+    )
+
+    depends_on('openmpi', when='mpi=openmpi')
+    depends_on('mpich', when='mpi=mpich')
+    depends_on('mvapich2', when='mpi=mvapich')
```

Comment:
This should be reverted to the original "virtual" `mpi` dependency. You can specify the specific MPI implementation either by default in the `packages.yaml` file, or explicitly for a single spec by `gpi-2 ^mpich`.

Reply:
Hi @sethrj,

Thanks for taking the time to revise the changes. I have updated the script following your suggestions, in particular we go back to this virtual dependency.